### DISCUSSION TITLE
invoke function take long time

### DISCUSSION THREAD

**Discussion by Xiaobaishushu25** [tauri-apps/tauri#8172] [Q&A]
I need to perform a file operation, and using writeBinaryFile will block the UI thread (does it require JavaScript multithreading?). I am not familiar with the frontend, so I want to pass the file to rust.
My process is: frontend retrieves the file → converts it to Uint8Array → invokes rust for processing.
My problem is that I found that the invoke function takes too long.
this is my  frontend code：
```ts
  let currentDateTime = new Date();
  console.log("start time: " + currentDateTime);
  await invoke("append_chunk_to_file", {path: "G:\\待处理文件\\11.png", chunk: chunk,end:true})
  currentDateTime = new Date();
  console.log("end time: " + currentDateTime);
```
this is my rust code,to avoid blocking threads, I used asynchronous operations:
```rust
#[tauri::command]
async fn append_chunk_to_file(window:Window,path: String, chunk: Vec<u8>,end:bool) -> Result<(), String> {
    let current_time = Local::now().time();
    println!("enter rust time: {}", current_time);
    println!("start{:?}",Instant::now());
    tokio::spawn(async move{
         //file operation
    });
    let instant = Instant::now();
    println!("end{:?}",instant);
    let current_time = Local::now().time();
    println!("return time: {}", current_time);
    Ok(())
}
```
this is run capture（the file is about 23mb），you can see that I called the rust function on the front end and it took me ten seconds to start executing the rust function.why does this take so long? Is there any solution?
![image](https://github.com/tauri-apps/tauri/assets/76581880/490f856b-d54d-4936-9524-022632100ef9)

**Comment by ZingerLittleBee** [comment]
Any news? the same problem

**Reply by Xiaobaishushu25** [reply]
用tarui2，有新的ipc
